src/amr/data/field/refine/magnetic_refine_patch_strategy.hpp

```diff
@@ -67,93 +62,98 @@ class MagneticRefinePatchStrategy : public SAMRAI::xfer::RefinePatchStrategy
     {
     }
 
-    // We compute the values of the new fine magnetic faces using what was already refined, ie the
-    // values on the old coarse faces.
+    // We compute the values of the new fine magnetic faces using what was already refined, ie
+    // the values on the old coarse faces.
     void postprocessRefine(SAMRAI::hier::Patch& fine, SAMRAI::hier::Patch const& coarse,
                            SAMRAI::hier::Box const& fine_box,
                            SAMRAI::hier::IntVector const& ratio) override
     {
         assertIDsSet();
 
-        auto& bx = FieldDataT::getField(fine, bx_id_);
-        auto& by = FieldDataT::getField(fine, by_id_);
-        auto& bz = FieldDataT::getField(fine, bz_id_);
+        auto& fields       = TensorFieldDataT::getFields(fine, b_id_);
+        auto& [bx, by, bz] = fields;
 
         auto layout        = PHARE::amr::layoutFromPatch<gridlayout_type>(fine);
         auto fineBoxLayout = Geometry::layoutFromBox(fine_box, layout);
 
-        SAMRAI::hier::Box fine_box_x
-            = Geometry::toFieldBox(fine_box, bx.physicalQuantity(), fineBoxLayout);
-        SAMRAI::hier::Box fine_box_y
-            = Geometry::toFieldBox(fine_box, by.physicalQuantity(), fineBoxLayout);
-        SAMRAI::hier::Box fine_box_z
-            = Geometry::toFieldBox(fine_box, bz.physicalQuantity(), fineBoxLayout);
+        auto fine_field_box = core::for_N<N, core::for_N_R_mode::make_array>([&](auto i) {
```

Comment:
you might find it simpler to use this https://github.com/PhilipDeegan/PHARE/blob/mkn/src/core/utilities/types.hpp#L673-L677